benchmarks/src/copy.cpp

```diff
@@ -10,8 +14,7 @@ static void handwritten_loop(benchmark::State& state) {
     const size_t r0 = static_cast<size_t>(state.range(0));
     std::vector<Contained> in_buffer(r0);
     std::vector<Contained> out_buffer(r0);
-    for (auto _ : state) {
-        (void) _;
+    for (auto [[maybe_unused]] _ : state) {

```

Comment:
These need to be `[[maybe_unused]] auto _`, per the grammar of [_for-range-declaration_](https://eel.is/c++draft/stmt.iter.general#nt:for-range-declaration). (https://godbolt.org/z/Gfr4snx1o) (Several places.) Where you have the attribute, it appertains to the type `auto` which currently has no meaning (https://godbolt.org/z/Pvrsc6zoc).